### DISCUSSION TITLE
What is the best way to use `usePrepareContractWrite` more than one time?

### DISCUSSION THREAD

**Discussion by sanjaybip** [wevm/wagmi#2024] [Q&A]
I have 2 buttons one is staking and other is withdrawing on same page adjacent to each other. Both calling same contract, but with different function and different argument value. 
Is it possible to define `usePrepareContractWrite` and `useContractWrite ` only one time but providing different function name with their argument in the click event handler of these 2 buttons? If possible then how? Or I have to define both of them 2 times for my use case?

**Comment by jeyemPF** [comment]
Yes, you can achieve this by defining a single instance of usePrepareContractWrite and useContractWrite hooks and then dynamically passing different function names and arguments based on the button clicked. Here's how you can do it:

`import { usePrepareContractWrite, useContractWrite } from 'yourContractLibrary';

const YourComponent = () => {
  // Define a single instance of usePrepareContractWrite
  const prepareContractWrite = usePrepareContractWrite();

  // Define a single instance of useContractWrite
  const contractWrite = useContractWrite();

  // Function to handle staking
  const handleStaking = async () => {
    const { args, contractFunction } = prepareContractWrite('stakingFunctionName', stakingArguments);
    await contractWrite(contractFunction, { args });
  };

  // Function to handle withdrawing
  const handleWithdrawing = async () => {
    const { args, contractFunction } = prepareContractWrite('withdrawingFunctionName', withdrawingArguments);
    await contractWrite(contractFunction, { args });
  };

  return (
    <div>
      <button onClick={handleStaking}>Stake</button>
      <button onClick={handleWithdrawing}>Withdraw</button>
    </div>
  );
};`
